core/src/main/java/org/apache/accumulo/core/client/ConnectionInfo.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.core.client;
+
+public interface ConnectionInfo {
+
+}
```

Comment:
Is this unfinished? If not, should have all the required methods here.

Reply:
`ConnectionInfo` lacks methods as I am not sure what to expose to users.  I don't want to expose most of the methods in `ConnectionInfoImpl`. However, I would say `ConnectionInfo` is not complete as it should have some methods for users to get basic information on their connection and all client properties.

Reply:
I added methods in 705d604759f9495cb